secant/Features/Scan/UIKitBridge/ScanUIView.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  ScanUIView.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 16.05.2022.
+//
+
+import Foundation
+import UIKit
+import AVFoundation
+
+public class ScanUIView: UIView {
+    var captureSession: AVCaptureSession?
+    var continueVideoSessionAfterScanning = false
+    
+    /// Rect of interest = area of the camera view used to try to recognize the qr codes.
+    private var internalRectOfInterest = CGRect(x: 0.0, y: 0.0, width: 1.0, height: 1.0)
+    var rectOfInterest: CGRect {
+        get { internalRectOfInterest }
+        set {
+            internalRectOfInterest = newValue
+            doInitialSetup()
+        }
+    }
+
+    var onQRScanningDidFail: (() -> Void)?
+    var onQRScanningSucceededWithCode: ((String?) -> Void)?
+    var onQRScanningDidStop: (() -> Void)?
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+    
+    override public class var layerClass: AnyClass {
+        return AVCaptureVideoPreviewLayer.self
+    }
+    
+    override public var layer: AVCaptureVideoPreviewLayer {
+        return super.layer as? AVCaptureVideoPreviewLayer ?? AVCaptureVideoPreviewLayer()
+    }
+}
+
+extension ScanUIView {
+    func stopScanning() {
+        captureSession?.stopRunning()
+        onQRScanningDidStop?()
+    }
+    
+    private func doInitialSetup() {
+        clipsToBounds = true
+        captureSession = AVCaptureSession()
+        
+        guard let videoCaptureDevice = AVCaptureDevice.default(for: .video) else {
+            return
+        }
+        
+        let videoInput: AVCaptureDeviceInput
+        do {
+            videoInput = try AVCaptureDeviceInput(device: videoCaptureDevice)
+        } catch {
+            return
+        }
+        
+        if captureSession?.canAddInput(videoInput) ?? false {
+            captureSession?.addInput(videoInput)
+        } else {
+            scanningDidFail()
+            return
+        }
+        
+        let metadataOutput = AVCaptureMetadataOutput()
+        
+        if captureSession?.canAddOutput(metadataOutput) ?? false {
+            captureSession?.addOutput(metadataOutput)
+            
+            metadataOutput.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+            metadataOutput.metadataObjectTypes = [.qr]
+        } else {
+            scanningDidFail()
+            return
+        }
+        
+        self.layer.session = captureSession
+        self.layer.videoGravity = .resizeAspectFill
+        
+        captureSession?.commitConfiguration()
+        captureSession?.startRunning()
+        metadataOutput.rectOfInterest = internalRectOfInterest
+    }
+    
+    func scanningDidFail() {
+        onQRScanningDidFail?()
+        captureSession = nil
+    }
+    
+    func found(code: String) {
+        onQRScanningSucceededWithCode?(code)
```

Comment:
I think the optional on `onQRScanningSucceededWithCode` can be changed by a concrete String instead.